### PR TITLE
Add new data sources hcp_packer_image and hcp_packer_iteration

### PR DESCRIPTION
**Pull request by SwampDragons**
The new data sources split the functionality from packer_image_iteration in a way that returns a less nested data structure that is easier for users to query and reason about.

### :ship: Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-hcp/blob/main/CHANGELOG.md):

```release-note
* **New data source** `packer_image` ([#194](https://github.com/hashicorp/terraform-provider-hcp/pulls/104)) in **private beta**. This data source works with the packer_iteration data source to replace the packer_image_iteration data source
* **New data source** `packer_iteration` ([#194](https://github.com/hashicorp/terraform-provider-hcp/pulls/104)) in **private beta**. This data source works with the packer_image data source to replace the packer_image_iteration data source
```

### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing: (Help! Tests are failing and I can't figure out why)

```
$make testacc TESTARGS='-run=TestAcc_dataSourcePacker -v'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAcc_dataSourcePacker -v -timeout 120m
?   	github.com/hashicorp/terraform-provider-hcp/internal/clients	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAcc_dataSourcePacker
--- PASS: TestAcc_dataSourcePacker (7.58s)
=== RUN   TestAcc_dataSourcePackerImage
--- PASS: TestAcc_dataSourcePackerImage (7.62s)
=== RUN   TestAcc_dataSourcePackerIteration
--- PASS: TestAcc_dataSourcePackerIteration (6.78s)
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/provider	22.462s
```

### CODE REVIEW

**Review by bcmdarroch — APPROVED** [review]
Looks good! Had a few comments about IDs and optional inputs, but should be easy to resolve.

File: internal/provider/data_source_packer_image.go [hunk 1]
```diff
@@ -132,5 +134,9 @@ func dataSourcePackerImageRead(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
+	if !found {
+		return diag.Errorf("Unable to load image with region %s and cloud %s for iteration %s.", region, cloudProvider, iterationID)
```
Comment (bcmdarroch) [inline internal/provider/data_source_packer_image.go:138]: Should this error change if there's no cloud provider passed?

File: internal/provider/data_source_packer_image.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"bucket_name": {
+				Description:      "The slug of the HCP Packer Registry image bucket to pull from.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"cloud_provider": {
+				Description: "Name of the cloud provider this image is stored-in, if any.",
```
[the remark this answers is not in the document]
Comment (SwampDragons) [reply internal/provider/data_source_packer_image.go:32]: Ah, this is copypasta -- this field is required. I'll fix.

File: internal/provider/data_source_packer_image.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+				Description: "Creation time of this build.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"build_id": {
+				Description: "HCP ID of this build.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"id": {
```
[the remark this answers is not in the document]
Comment (SwampDragons) [reply internal/provider/data_source_packer_image.go:72]: I was wondering. I think that makes sense.

File: templates/data-sources/packer_image.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+-> **Note:** This feature is currently in private beta.
+
+{{ .Description | trimspace }}
+
+## Example Usage
+
+{{ tffile "examples/data-sources/hcp_packer_image/data-source.tf" }}
+
+## Schema
```
[the remark this answers is not in the document]
Comment (SwampDragons) [reply templates/data-sources/packer_image.md.tmpl:18]: oh cool! I'll give that a try.

File: internal/provider/data_source_packer_image.go (hunk 1 above)
Comment (SwampDragons) [reply internal/provider/data_source_packer_image.go:138]: see above; the cloud provider must always be passed. It was a docs mistake.

**Comment by SwampDragons** [comment]
Thanks for taking the time to review! I'll try to fix these and push updates tomorrow.

**Comment by SwampDragons** [comment]
@bcmdarroch I addressed your changes; want to re-review or can I merge as is?

**Comment by SwampDragons** [comment]
Merging since it had an approval and it turns out Brenna is on vacation.

**Pull request merged**